src/customer/strategies/googlepay/googlepay-braintree-customer-strategy.ts

```diff
@@ -0,0 +1,120 @@
+
+export default class GooglePayBraintreeCustomerStrategy extends CustomerStrategy {
+    private _walletButton?: HTMLElement;
+
+    constructor(
+        store: CheckoutStore,
+        private _remoteCheckoutActionCreator: RemoteCheckoutActionCreator,
+        private _googlePayPaymentProcessor: GooglePayPaymentProcessor,
+        private _formPoster: FormPoster
+    ) {
+        super(store);
+    }
+
+    initialize(options: CustomerInitializeOptions): Promise<InternalCheckoutSelectors> {
+        if (this._isInitialized) {
+            return super.initialize(options);
+        }
+
+        const { googlepaybraintree, methodId }  = options;
+
+        if (!googlepaybraintree || !methodId) {
+            throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
+        }
+
+        return this._googlePayPaymentProcessor.initialize(methodId)
+            .then(() => {
+                const walletButton = this._createSignInButton(googlepaybraintree.container);
+
+                if (walletButton) {
+                    this._walletButton = walletButton;
+                    this._walletButton.addEventListener('click', this._handleWalletButtonClick);
+                }
+            })
+            .then(() => super.initialize(options));
+    }
+
+    deinitialize(options?: CustomerRequestOptions): Promise<InternalCheckoutSelectors> {
+        if (!this._isInitialized) {
+            return super.deinitialize(options);
+        }
+
+        if (this._walletButton && this._walletButton.parentNode) {
+            this._walletButton.parentNode.removeChild(this._walletButton);
+            this._walletButton.removeEventListener('click', this._handleWalletButtonClick);
+            this._walletButton = undefined;
+        }
+
+        return this._googlePayPaymentProcessor.deinitialize()
+            .then(() => super.deinitialize(options));
+    }
+
+    signIn(credentials: CustomerCredentials, options?: CustomerRequestOptions): Promise<InternalCheckoutSelectors> {
+        throw new NotImplementedError(
+            'In order to sign in via Google Pay, the shopper must click on "Google Pay" button.'
+        );
+    }
+
+    signOut(options?: CustomerRequestOptions): Promise<InternalCheckoutSelectors> {
+        const state = this._store.getState();
+        const payment = state.payment.getPaymentId();
+
+        if (!payment) {
+            return Promise.resolve(this._store.getState());
+        }
+
+        return this._store.dispatch(
+            this._remoteCheckoutActionCreator.signOut(payment.providerId, options)
+        );
+    }
+
+    private _createSignInButton(containerId: string): HTMLElement {
+        const container = document.querySelector(`#${containerId}`);
+
+        if (!container) {
+            throw new InvalidArgumentError('Unable to create sign-in button without valid container ID.');
+        }
+
+        const button = this._googlePayPaymentProcessor.createButton(() => {});
```

Comment:
Could you please take a look at my comment regarding the `onClick` handler and Google's recommended usage?